plugin/plugin.go

```diff
@@ -67,47 +71,60 @@ type Args struct {
 
 	// connect hostname (required)
 	ConnectHostname string `envconfig:"PLUGIN_CONNECT_HOSTNAME"`
+	IssueKeys []string `envconfig:"PLUGIN_ISSUEKEYS"`
 }
 
 // Exec executes the plugin.
 func Exec(ctx context.Context, args Args) error {
 	var (
-		environ  = toEnvironment(args)
-		issue    = extractIssue(args)
+		environ         = toEnvironment(args)
+		environmentID   = toEnvironmentId(args)
+		environmentType = toEnvironmentType(args)
+		issue    string
+		issues   []string
 		state    = toState(args)
 		version  = toVersion(args)
 		deeplink = toLink(args)
 	)
 
+	// ExtractInstanceName extracts the instance name from the provided URL if any
+	instanceName := ExtractInstanceName(args.Instance)
+
 	logger := logrus.
 		WithField("client_id", args.ClientID).
 		WithField("cloud_id", args.CloudID).
 		WithField("project_id", args.Project).
-		WithField("instance", args.Instance).
+		WithField("instance", instanceName).
 		WithField("pipeline", args.Name).
 		WithField("environment", environ).
 		WithField("state", state).
-		WithField("version", version)
+		WithField("environment Type", environmentType).
+		WithField("environment ID", environmentID)
 
-	if issue == "" {
-		logger.Debugln("cannot find issue number")
-		return errors.New("failed to extract issue number")
+	if len(args.IssueKeys) > 0 {
+		issues = args.IssueKeys
+	} else {
+		issue = extractIssue(args)
+		if issue == "" {
+			logger.Debugln("cannot find issue number")
+			return errors.New("failed to extract issue number")
+		}
+		logger = logger.WithField("issue", issue)
 	}
```

Comment:
let's not duplicate how we handle both the array (issues) and the single issue (issue [old logic]) separately, and keep the logic simple:

// check if PLUGIN_ISSUEKEYS is provided
if len(args.IssueKeys) > 0 {
    issues = args.IssueKeys
} else {
    // fallback to extracting from commit if no issue keys are passed
    issue = extractIssue(args)
    if issue == "" {
        logger.Debugln("cannot find issue number")
        return errors.New("failed to extract issue number")
    }
    issues = []string{issue}  // add the single issue here for consistency
}

and modify the deployment payload accordingly.

Reply:
Initially I have done this way, but observed one issue that if PLUGIN_ISSUEKEYS are not provided then Issue keys will be  populated by old way(extractIssue) right. That means deployment payload contains always both Issuekeys and associations like below. With this payload when i tested restapi (postman) I was getting error in the response as below ("message": "issueKeys and associations are mutually exclusive.) Is this expected.? Also it could be existing behaviour. But will this cause any issue?  
Request chunk:
"issueKeys": [
        "SCRUM-1"
      ],
      "associations": [
        {
          "associationType": "issueIdOrKeys",
          "values": [
            "SCRUM-1"
          ]
        }
      ],

Response chunk:
"errors": [
                {
                    "message": "issueKeys and associations are mutually exclusive. Either only specify issueKeys or pass issueKeys as an associationType.",
                    "key": "deploymentdata.issueKeysOrAssociationsOrNone.invalid"
                },